engine/src/main/java/io/zeebe/engine/processing/bpmn/behavior/BpmnStateTransitionBehavior.java

```diff
@@ -84,7 +84,8 @@ public BpmnElementContext transitionToActivating(final BpmnElementContext contex
       }
 
       if (context.getElementInstanceKey() == -1) {
-        // processing ACTIVATE_ELEMENT command and no key was generated before
+        // Some elements (such as multi instance) need to generate the key before the
+        // ACTIVATE command, to prepare the state (e.g. set variables).
```

Comment:
Oops, sorry, I didn't think you'd paste it as is :sweat_smile: 

Here we want to tell some unsuspecting future developer why we're checking for -1 - why is it sometimes already a number, and why it's -1. In this case, we just explained why sometimes it's a number, but not why -1.

So we probably want to explain that "Sometimes elements already have a key before activating in order to prepare state that must be present when activating (e.g. set multi instance body specific variables), but most of the times they do not and we must generate one"

Also, nit pick (feel free to ignore), I would also suggest swapping the conditions, actually (and keeping the comment). e.g.

```java
      // Sometimes elements already have a key before activating in order to prepare state that must be
      // present when activating (e.g. set multi instance body specific variables), but most of the times they do not and 
      // we must generate one
      if (context.getElementInstanceKey() > -1) {
        return transitionTo(context, ProcessInstanceIntent.ELEMENT_ACTIVATING);
      }

      final var newElementInstanceKey = keyGenerator.nextKey();
      final var newContext =
          context.copy(
              newElementInstanceKey,
              context.getRecordValue(),
              ProcessInstanceIntent.ELEMENT_ACTIVATING);
      return transitionTo(newContext, ProcessInstanceIntent.ELEMENT_ACTIVATING);
    }
```

Reply:
tbh I would expect that this is only a short period of time that we need this. at the end I would expect all ACTIVATE commands are written without a key.